src/components/calculator.js

```diff
@@ -4,68 +4,57 @@ import Buttons from './buttons';
-        <Buttons name="eight" label="8" buttonClick={(string) => this.update(string)} />
-        <Buttons name="nine" label="9" buttonClick={(string) => this.update(string)} />
-        <Buttons name="multiply" label="x" buttonClick={(string) => this.update(string)} />
-        <Buttons name="divide" label="÷" buttonClick={(string) => this.update(string)} />
-        <Buttons name="add" label="+" buttonClick={(string) => this.update(string)} />
-        <Buttons name="subtract" label="-" buttonClick={(string) => this.update(string)} />
-        <Buttons name="equal" label="=" buttonClick={(string) => this.update(string)} />
-        <Buttons name="decimal" label="." buttonClick={(string) => this.update(string)} />
-        <Buttons name="all-clear" label="AC" buttonClick={(string) => this.update(string)} />
-        <Buttons name="positive-negative" label="+/-" buttonClick={(string) => this.update(string)} />
-        <Buttons name="percent" label="%" buttonClick={(string) => this.update(string)} />
-      </div>
-    );
-  }
-}
+    }, buttonPress);
+    setTotal(object.total);
+    setOperation(object.operation);
+    setNext(object.next);
+  };
 
-export default Calculator;
+  return (
+    <div className="grid-container">
+      <Answer name="answer" previous={total} operand={operation} current={next} />
+      <Buttons name="zero" label="0" buttonClick={(string) => update(string)} />
+      <Buttons name="one" label="1" buttonClick={(string) => update(string)} />
+      <Buttons name="two" label="2" buttonClick={(string) => update(string)} />
+      <Buttons name="three" label="3" buttonClick={(string) => update(string)} />
+      <Buttons name="four" label="4" buttonClick={(string) => update(string)} />
+      <Buttons name="five" label="5" buttonClick={(string) => update(string)} />
+      <Buttons name="six" label="6" buttonClick={(string) => update(string)} />
+      <Buttons name="seven" label="7" buttonClick={(string) => update(string)} />
+      <Buttons name="eight" label="8" buttonClick={(string) => update(string)} />
+      <Buttons name="nine" label="9" buttonClick={(string) => update(string)} />
+      <Buttons name="multiply" label="x" buttonClick={(string) => update(string)} />
+      <Buttons name="divide" label="÷" buttonClick={(string) => update(string)} />
+      <Buttons name="add" label="+" buttonClick={(string) => update(string)} />
+      <Buttons name="subtract" label="-" buttonClick={(string) => update(string)} />
+      <Buttons name="equal" label="=" buttonClick={(string) => update(string)} />
+      <Buttons name="decimal" label="." buttonClick={(string) => update(string)} />
+      <Buttons name="all-clear" label="AC" buttonClick={(string) => update(string)} />
+      <Buttons name="positive-negative" label="+/-" buttonClick={(string) => update(string)} />
+      <Buttons name="percent" label="%" buttonClick={(string) => update(string)} />
```

Comment:
- [ ] (Optional) I would like to advise you to create an array of objects. Each object in the array will have keys for `name` and `label`. You can then map through the array, restructure the keys in the parameter and then pass them as props to the `Buttons` components. You can read this [article](https://www.geeksforgeeks.org/how-to-map-data-into-components-using-reactjs/) to have an idea of what I am talking about. 👍